src/app/data-model/BUILD.gn

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import("//build_overrides/chip.gni")
+
+source_set("data-model") {
+  sources = [
+    "BasicTypes.h",
+    "DecodableList.h",
+    "Decode.h",
+    "Encode.h",
+    "FabricScoped.h",
+    "List.h",
+    "NullObject.h",
+    "Nullable.h",
+    "TagBoundEncoder.h",
+    "WrappedStructEncoder.h",
+  ]
+
+  deps = [
+    # TODO: dependencies NOT declared because they are not part of
+    #       any GN dependency. Overall src/app seems to suffer greatly
+    #       of this, in part due to zap-generated code dependency.
+    #
+    #  - app/util/attribute-storage-null-handling.h
+    #  - app/ConcreteAttributePath.h
+    #
+    "${chip_root}/src/app/common:enums",
```

Comment:
I don't get this part.  Nothing in the sources list above includes cluster-enums.h, right?  Why is it a dep here?

Reply:
I will re-check however a large part of this was done by "added enums dependency" followed by compile errors where gn told me where the deps needed adding.

Reply:
this seems required:

```
src/app/data-model/Decode.h
21:#include <app-common/zap-generated/cluster-enums-check.h>
```

Reply:
Ah, cluster-enums-check is in there.  OK, that makes sense.  I had missed that.